bsb_nest/connection.py

```diff
@@ -95,7 +97,11 @@ def predict_mem_iterator(self, pre_nodes, post_nodes, cs):
         predicted_all_mem = (
             len(pre_nodes) * 8 * 2 + len(post_nodes) * 8 * 2 + len(cs) * 6 * 8 * (16 + 2)
         ) * MPI.get_size()
-        predicted_local_mem = predicted_all_mem / len(cs.get_local_chunks("out"))
+        predicted_local_mem = (
+            (predicted_all_mem / len(cs.get_local_chunks("out")))
+            if len(cs.get_local_chunks("out")) > 0
+            else 0.0
+        )
```

Comment:
Iterating over the chunks in an HDF5 file seems incredibly slow. Do it only once:

```suggestion
        n_chunks = len(cs.get_local_chunks("out"))
        predicted_local_mem = (
            (predicted_all_mem / n_chunks)
            if n_chunks > 0
            else 0.0
        )
```

I'm not sure if this applies here, it's just a precaution.